### ISSUE TITLE
pandoc complains about missing webSettings.xml

### ISSUE THREAD

**Issue by aignas** [jgm/pandoc#1185]
If I output .md to .docx, edit it with LibreOffice and then try to use it as a reference docx, pandoc complains:

```
pandoc: user error (word/webSettings.xml missing in reference docx)
```

Is this a bug in LibreOffice, or is there something pandoc could do about this?

Thanks

**Comment by curiositry** [comment]
I’m having the same problem @aignas had. I’m running pandoc 1.16.0.2. I created a .docx with pandoc, and then modified it with LibreOffice and am attempting to use it as my reference.docx. I get the error: 

```
pandoc: user error (word/webSettings.xml missing in reference docx)
```

… which I thought was resolved in v1.12.4. Any suggestions?

**Comment by curiositry** [comment]
I’m having the same problem @aignas had. I created a .docx with pandoc, and then modified it with LibreOffice and am attempting to use it as my reference.docx. I’m running pandoc 1.16.0.2., and I get the following error:

pandoc: user error (word/webSettings.xml missing in reference docx)

… which I thought was resolved in v1.12.4. Any suggestions @jgm?

**Comment by mb21** [comment]
@curiositry you should update to the newest pandoc version (http://pandoc.org/installing.html) and report back if you still experience the issue..

**Comment by jgm** [comment]
+++ Curiositry [Feb 21 17 19:12 ]:
>   I’m having the same problem [1]@aignas had. I created a .docx with
>   pandoc, and then modified it with LibreOffice and am attempting to use
>   it as my reference.docx. I’m running pandoc 1.16.0.2., and I get the
>   following error:
>
>   pandoc: user error (word/webSettings.xml missing in reference docx)
>
>   … which I thought was resolved in v1.12.4. Any suggestions [2]@jgm?

I suggest you try the latest release.

**Comment by curiositry** [comment]
@jgm, @mb21 Thanks for responding. Unfortunately, I get the same error with pandoc v1.19.2.1.

**Comment by jgm** [comment]
This may have to do with the way you installed pandoc.
How did you install it?  Did you use one of our binary
packages, compile it yourself, or what?

What happens if you do

    pandoc -D latex

**Comment by curiositry** [comment]
Thanks @jgm — I originally installed Pandoc 1.16.0.2 from the Ubuntu repositories. When you suggested I use the latest version, I installed the Pandoc 1.19.2.1 DEB from the GitHub releases page. Could leftovers from the Ubuntu version have messed things up for the new version?

`pandoc -D latex` outputs LaTeX starting with `\documentclass[$if(fontsize)$$fontsize$,…` and ending with `\end{document}`. (it looks like Pandoc’s LaTeX template to me, but I could be wrong; I’m not too familiar with LaTeX)

**Comment by curiositry** [comment]
I don’t know if this is relevant, but, curiously, while `pandoc -D odt` outputs XML I assume to be the contents of the Pandoc template, `pandoc -D docx` produces nothing, and `pandoc -D doc` produces the error: `pandoc: Could not find data file templates/default.doc`

**Comment by jgm** [comment]
OK, so that means at least it's finding your templates.

Try:

    pandoc --print-default-data-file docx/word/webSettings.xml

Also check to make sure you don't have anything a
~/.pandoc/docx directory.

**Comment by curiositry** [comment]
That command returns:

```
<?xml version="1.0" encoding="UTF-8"?>
<ns0:webSettings xmlns:ns0="http://schemas.openxmlformats.org/wordprocessingml/2006/main">
  <ns0:allowPNG />
  <ns0:doNotSaveAsSingleFile />
</ns0:webSettings>
```

And there is no `~/.pandoc/docx` directory on my system.

Thanks for helping out!

**Comment by jgm** [comment]
Weird.  That means that pandoc finds the data file.
OK. I need to think more about what might be causing this for you.

+++ Curiositry [Feb 23 17 12:45 ]:

**Comment by jgm** [comment]
Just to confirm: you created the reference.docx with the
same version of pandoc as your're running -- it's not an
older one?

I suppose it's possible that LibreOffice is stripping out
the word/webSettings.xml component for some reason.

**Comment by jgm** [comment]
Oh, and do you have a reference.docx in your ~/.pandoc
directory?  If so, move it somewhere else.

+++ Curiositry [Feb 23 17 11:51 ]:

**Comment by curiositry** [comment]
No, I created my reference.docx  with the version of Pandoc I installed from the Ubuntu repos; but the issue only appeared after I edited it with LibreOffice to conform to standard manuscript formatting. 

I could try making a new one based on a document created with my current Pandoc installation, but it appears to be an issue with LibreOffice messing up the webSettings.xml, and Pandoc not falling back on a default.

**Comment by curiositry** [comment]
When I remove the reference.docx, docx conversion works. When I put it back, it breaks. When I put a docx generated with the latest version of Pandoc in `.pandoc`, and name it reference.docx, it doesn’t break it; but when I open it with LibreOffice, make some small change, and save it, the webSettings error reappears.

So it seems that:

1. The webSettings error is caused by LibreOffice
2. The Pandoc version the reference document was originally created with doesn’t matter.

**Comment by jgm** [comment]
This is puzzling, because the code is supposed to insert
webSettings.xml from the system default reference.docx if it is
not included in the reference.docx you supply. (This was the
fix to the issue above.)  It seems this isn't working, but
I don't understand why.

Can you link to, attach or send a copy of the problematic
reference.docx so I can test?

+++ Curiositry [Feb 23 17 13:10 ]:

**Comment by curiositry** [comment]
Certainly. Here it is:

[reference.docx](https://github.com/jgm/pandoc/files/801486/reference.docx)

**Comment by jgm** [comment]
@curiositry I had no trouble using that reference.docx.  (I tried MANUAL.txt from pandoc with it, and it worked fine; I got a different font and line spacing than usual so I know it is using the reference.docx.)

I tried with both HEAD and 1.19.1 release.

**Comment by curiositry** [comment]
@jgm I purged my Pandoc install and started fresh with the latest stable from the releases page. This changed the error from `webSettings.xml` to `numbering.xml`, which could be called progress. I tried a fresh install of the latest version of LibreOffice, and I tried opening it in Word and saving it — neither of which had any effect.

I ended up  adding `webSettings.xml` and `numbering.xml` to the archive by hand. This is a bit inconvenient, but it works. If you have any ideas about why it isn’t falling back on Pandoc’s `numbering.xml`, that would be great, but no pressure: you’ve been very helpful already and I have a workaround that works for now.

Thanks for making an amazing piece of software, and taking the time to support it!
